pkg/sql/schemachanger/testdata/alter_table_add_column

```diff
@@ -2664,3 +2664,1454 @@ update progress of schema change job #1: "all stages completed"
+     id: 107
+  -  modificationTime:
+  -    wallTime: "1640995200000000006"
+  +  modificationTime: {}
+     name: sq1
+     parentId: 104
+  ...
+       start: "1"
+     unexposedParentSchemaId: 105
+  -  version: "12"
+  +  version: "13"
+adding table for stats refresh: 106
+update progress of schema change job #1: "PostCommitPhase stage 8 of 13 with 1 BackfillType op pending"
+commit transaction #9
+begin transaction #10
+## PostCommitPhase stage 8 of 13 with 1 BackfillType op
+backfill indexes [10] from index #8 in table #106
+commit transaction #10
+begin transaction #11
+## PostCommitPhase stage 9 of 13 with 4 MutationType ops
+upsert descriptor #106
+  ...
+       - PUBLIC
+       - WRITE_ONLY
+  -    - BACKFILL_ONLY
+  +    - DELETE_ONLY
+       - ABSENT
+       - WRITE_ONLY
+  ...
+     formatVersion: 3
+     id: 106
+  -  modificationTime:
+  -    wallTime: "1640995200000000009"
+  +  modificationTime: {}
+     mutations:
+     - direction: ADD
+  ...
+         version: 4
+       mutationId: 1
+  -    state: BACKFILLING
+  +    state: DELETE_ONLY
+     name: tbl
+     nextColumnId: 6
+  ...
+       time: {}
+     unexposedParentSchemaId: 105
+  -  version: "27"
+  +  version: "28"
+upsert descriptor #107
+  ...
+     formatVersion: 3
+     id: 107
+  -  modificationTime:
+  -    wallTime: "1640995200000000009"
+  +  modificationTime: {}
+     name: sq1
+     parentId: 104
+  ...
+       start: "1"
+     unexposedParentSchemaId: 105
+  -  version: "13"
+  +  version: "14"
+update progress of schema change job #1: "PostCommitPhase stage 10 of 13 with 1 MutationType op pending"
+commit transaction #11
+begin transaction #12
+## PostCommitPhase stage 10 of 13 with 4 MutationType ops
+upsert descriptor #106
+  ...
+       - PUBLIC
+       - WRITE_ONLY
+  -    - DELETE_ONLY
+  +    - MERGE_ONLY
+       - ABSENT
+       - WRITE_ONLY
+  ...
+     formatVersion: 3
+     id: 106
+  -  modificationTime:
+  -    wallTime: "1640995200000000011"
+  +  modificationTime: {}
+     mutations:
+     - direction: ADD
+  ...
+         version: 4
+       mutationId: 1
+  -    state: DELETE_ONLY
+  +    state: DELETE_AND_WRITE_ONLY
+     name: tbl
+     nextColumnId: 6
+  ...
+       time: {}
+     unexposedParentSchemaId: 105
+  -  version: "28"
+  +  version: "29"
+upsert descriptor #107
+  ...
+     formatVersion: 3
+     id: 107
+  -  modificationTime:
+  -    wallTime: "1640995200000000011"
+  +  modificationTime: {}
+     name: sq1
+     parentId: 104
+  ...
+       start: "1"
+     unexposedParentSchemaId: 105
+  -  version: "14"
+  +  version: "15"
+update progress of schema change job #1: "PostCommitPhase stage 11 of 13 with 1 BackfillType op pending"
+commit transaction #12
+begin transaction #13
+## PostCommitPhase stage 11 of 13 with 1 BackfillType op
+merge temporary indexes [11] into backfilled indexes [10] in table #106
+commit transaction #13
+begin transaction #14
+## PostCommitPhase stage 12 of 13 with 1 ValidationType op
```

Comment:
aside that's irrelevant to this change: we should probably update the statuses in the descriptors between backfill and validation. I think the backfill progress will mean that we don't backfill again, but something is somewhat unsatisfying about going from MERGE_ONLY to PUBLIC for me.